app/controllers/submit.js

```diff
@@ -333,8 +338,38 @@ export default Ember.Controller.extend(BasicsValidations, NodeActionsMixin, Tagg
             this.send('next', this.get('_names.0'));
         },
         resetFileUpload() {
-            this.send('changeState', State.START);
-            this.set('fileAndNodeLocked', false);
+            var promisesArray = [];
+            var filePromises = [];
+            this.get('filesUploadedForPreprint').forEach((file) => {
+                filePromises.push(this.get('fileManager').deleteFile(file).then(() => {
+                    this.get('toast').info('File removed!');
+                }).catch(() => {
+                    this.get('toast').error('Could not remove uploaded file');
+                }));
+            });
+
+            Ember.RSVP.all(filePromises).then(() => {
+                this.get('projectsCreatedForPreprint').forEach((project) => {
+                    promisesArray.push(this.store.adapterFor('node').ajax(project.get('links.self'), 'DELETE').then(() => {
```

Comment:
We seem to be discussing two separate but related  things here...
- The method on the model object itself, [built into ember](http://emberjs.com/api/data/classes/DS.Model.html#method_deleteRecord)
- An [action](https://github.com/abought/ember-osf/blob/0f93704050569ad2cda033a81e4f3a9075a391c4/addon/mixins/node-actions.js#L68-L68) in the NodeActionsMixin, which uses the insta-save version of the method under the hood

I wasn't aware of anything in ember-osf that was straight-out breaking deletions. That's interesting to know- if it's not tracked yet, might be a good idea to make it so when you have a chance.